source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,761 @@
-         sudo swapoff -a
+* `CRI+O`_
+* `containerd`_
+* `Docker`_
 
-      .. note::
+The container runtime that you choose will dictate the steps necessary to
+initialize the master cluster with :command:`kubeadm init`.
 
-         On systems with limited resources, some performance degradation may
-         be observed while swap is disabled.
+CRI+O
+-----
 
-#. Switch to root to modify the `hosts` file:
+For information on CRI+O as a Kubernetes CRI, see `What is
+CRI+O?`_. To use CRI+O as the Kubernetes CRI:
+
+#. Start the CRI-O service and enable it to run at boot automatically:
 
    .. code-block:: bash
 
-      sudo -s
+      sudo systemctl enable --now crio.service
 
-#.  Create (or edit if it exists) the hosts file that Kubernetes will read to
-    locate the master's host:
+   When the crio service starts for the first time, it will create a
+   configuration file for crio at :file:`/etc/crio/crio.conf`.
 
-    .. code-block:: bash
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
 
-       echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
+   .. important:: 
 
-#.  Exit root:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. 
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is to use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
 
-    .. code-block:: bash
+   .. code-block:: bash
 
-       exit
+      sudo kubeadm init \
+      --cri-socket=unix:///run/crio/crio.sock \
+      --pod-network-cidr 10.244.0.0/16
 
-Configure and run Kubernetes
-****************************
 
-This section describes how to configure and run Kubernetes with CRI-O and kata-runtime. We will be using  the :command:`kubectl` command-line interface to control Kubernetes.
+#. (Optional) By default, CRI+O will use runc as the default
+   runtime. CRI+O can optionally provide Kata Containers as a runtime. See
+   the `Add the Kata runtime to Kubernetes`_ section for details.
 
-Configure and run CRI-O + kata-runtime
-======================================
+   With CRI+O, the `Kata Containers`_  can be set as the runtime with a
+   per-pod *RuntimeClass* annotation. 
 
-#.  Enable the CRI-O service:
+   .. note:: 
 
-    .. code-block:: bash
+      If you are using CRI-O + Kata Containers as the runtime and choose the
+      *flannel* for pod networking (see `Choose a pod network add-on`_), the
+      :file:`/etc/crio/crio.conf` file needs to include the value below. On
+      |CL| this is done automatically. 
 
-       sudo systemctl enable crio.service
+      .. code-block:: console
 
-#.  Enter these commands to restart the system services so that CRI-O will be restarted, and will automatically start with every reboot.
+         [crio.runtime]
+         manage_network_ns_lifecycle = true
```

Comment:
is is the default setting so no longer needed in this doc afaics: https://github.com/clearlinux-pkgs/cri-o/blob/master/make_args#L7